packages/website/utils/content.ts

```diff
@@ -86,18 +44,57 @@ async function fetchFiles(targetPath: string) {
   return fetchedFiles;
 }
 
-async function geltAllMDX() {
-  const allMDX = await fetchFiles(componentsPath);
-  const mdxData = allMDX.map((filepath) => getSlugByFilepath(filepath));
+async function getAllMdx(paths: string[]) {
+  let allMDX: string[] = [];
 
-  // Saving a map of slugs and the filepath to their mdx in a JSON file
-  const mdxFilepathBySlug = mdxData.reduce((acc, mdx) => {
-    return { ...acc, [mdx.slug]: mdx.filepath };
-  }, {});
-  const data = JSON.stringify(mdxFilepathBySlug, null, 2);
-  fs.writeFileSync('utils/mdxFilepathBySlug.json', data);
+  for (const path of paths) {
+    const newFiles = await fetchFiles(path);
+    allMDX = allMDX.concat(newFiles);
+  }
 
+  const mdxData = allMDX
+    .map((filepath) => getDataByFilepath(filepath))
+    .filter((data) => {
+      // return only those mdx files which have 'slug' set in meta section
+      return data.frontMatter.data.slug;
+    });
```

Comment:
Is it possible to move this filter to `getDataByFilepath`?

The way it is now, it looks like we are getting unnecessary data in the `map` method

Reply:
`getDataByFilepath` cannot filter things out because it's used to transform data in `map` method

Reply:
Ah! of course! 
And you can't return a null or something like that
Got it

In this case, I have another argument in favor of destructuring that frontMatter

**please tell me if I'm miss understanding this**

In getDataByFilepath, you return:

```
{
    content: fileContents,
    frontMatter: { data },
    filepath,
};
```

right?

and then here you need to do that filter, but if you destructure that you could do:
```suggestion
    .filter((data) => {
      // return only those mdx files which have 'slug' set in meta section
      return data.frontMatter.slug;
    });
```

even better:
```suggestion
    .filter(({ frontMatter }) => {
      // return only those mdx files which have 'slug' set in meta section
      return slug;
    });
```

no? Am I seeing things?